src/pymoca/backends/casadi/model.py

```diff
@@ -524,16 +524,44 @@ def simplify(self, options):
                         value = None
 
                     if variable is not None:
-                        del alg_states[variable.name()]
-
-                        variables.append(variable)
                         if eq.is_op(ca.OP_SUB):
-                            values.append(value)
+                            return variable, value
                         else:
-                            values.append(-value)
+                            return variable, -value
 
-                        # Skip this equation
-                        continue
+                if eq.n_dep() == 2 \
+                    and eq.is_op(ca.OP_ADD) \
+                    and eq.dep(0).is_op(ca.OP_IF_ELSE_ZERO) \
+                    and eq.dep(1).is_op(ca.OP_IF_ELSE_ZERO):
+                    # Once passed through _expand_simplify_mx, a CasADi if_else
+                    # is converted to a sum of two if_else_zero operations.
```

Comment:
Good comment. This should make it clear enough that this only applies in the expand_mx case (which is the only time expand_simplify_mx is called afaik).